k9mail-library/src/main/java/com/fsck/k9/mail/store/imap/SelectOrExamineResponse.java

```diff
@@ -33,6 +29,39 @@ public static SelectOrExamineResponse newInstance(List<ImapResponse> imapRespons
         return new SelectOrExamineResponse(imapResponses, folder);
     }
 
+    private static Boolean isModeReadWriteIfAvailable(ImapResponse imapResponse) {
+        if (!imapResponse.isList(1)) {
+            return null;
+        }
+
+        ImapList responseTextList = imapResponse.getList(1);
+        if (!responseTextList.isString(0)) {
+            return null;
+        }
+
+        String responseCode = responseTextList.getString(0);
+        if ("READ-ONLY".equalsIgnoreCase(responseCode)) {
+            return Boolean.FALSE;
+        } else if ("READ-WRITE".equalsIgnoreCase(responseCode)) {
+            return Boolean.TRUE;
+        }
+
+        return null;
+    }
+
+    private static boolean isResponseValid(List<ImapResponse> imapResponses) {
```

Comment:
did you do some "order static methods" in android studio? it's fine if you do it but please keep these types of changes in separate commits so they don't make review unnecessarily hard :)